engine/classes/Elgg/HooksRegistrationService.php

```diff
@@ -90,6 +90,33 @@ public function unregisterHandler($name, $type, $callback) {
 	}
 
 	/**
+	 * Removes all handlers
+	 *
+	 * @param string   $name
+	 * @param string   $type
+	 *
+	 * @return array
+	 * @access private
+	 */
+	public function clearHandlers($name, $type = null) {
```

Comment:
What I'm saying is that the user of this function will likely trying to prevent any handlers from running under the given hook/type. We should at least remind them that handlers registered using the keyword `all` may still apply to the hook.